src/plugins/Prism.ts

```diff
@@ -60,27 +63,42 @@ function getDecorations({ doc, name }) {
 
     const nodes = refractor.highlight(block.node.textContent, language);
 
-    flattenDeep(parseNodes(nodes))
-      .map((node: ParsedNode) => {
-        const from = startPos;
-        const to = from + node.text.length;
-
-        startPos = to;
-
-        return {
-          ...node,
-          from,
-          to,
-        };
-      })
-      .forEach(node => {
-        const decoration = Decoration.inline(node.from, node.to, {
-          class: (node.classes || []).join(" "),
+    if (!cache[block.pos] || !cache[block.pos].node.eq(block.node)) {
```

Comment:
We should move `refractor.highlight` inside the `if` statement, right?!

Reply:
Fixed